### ISSUE TITLE
ethermint 0.5.4 - Cannot run ethermint executable, errors with panic: runtime error: slice bounds out of range

### ISSUE THREAD

**Issue by emizzle** [cosmos/ethermint-archive#387]
## Installed ethermint and tendermint from brew
Followed instructions from [Docs > Install Ethermint](https://ethermint.readthedocs.io/en/master/getting-started/install.html), I installed ethermint and tendermint successfully three different ways (from brew, from binary, and from source). In all three instances, I encountered the exact same error. 

1. Installed from brew using:
```
brew update
brew tap tendermint/homebrew-tendermint
brew upgrade openssl
brew upgrade libressl
brew upgrade curl
brew install tendermint
brew install ethermint
```

2. Installed from source and built using:
```
go get -u -d github.com/tendermint/ethermint
go get -u -d github.com/tendermint/tendermint
cd $GOPATH/src/github.com/tendermint/ethermint
make get_vendor_deps (this step missing from doco)
make install
cd ../tendermint
make get_vendor_deps (this step missing from doco)
make install
```

3. Installed from binary by [downloading the binaries](https://tendermint.com/downloads)
    1. [ethermint_0.5.3_darwin-10.6-amd64.zip](https://github.com/tendermint/ethermint/releases/download/v0.5.3/ethermint_0.5.3_darwin-10.6-amd64.zip)
    1. [tendermint_0.15.0_darwin_amd64.zip](https://github.com/tendermint/tendermint/releases/download/v0.15.0/tendermint_0.15.0_darwin_amd64.zip)
    1. [abci 0.9.0 source zip](https://github.com/tendermint/abci/archive/v0.9.0.zip)

## After install, running `ethermint` produces the error
The same error occurred independently after each different installation attempt.

`tendermint init --home ~/.ethermint/tendermint`
Works

Running `ethermint` with *any* command args fails with:
```
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/tendermint/ethermint/vendor/github.com/ethereum/go-ethereum/cmd/utils.NewApp(0x4a46dc9, 0x5, 0x4a6540a, 0x24, 0xc4200c1708)
	/private/tmp/ethermint-20180215-18815-12utxc0/ethermint-0.5.4/src/github.com/tendermint/ethermint/vendor/github.com/ethereum/go-ethereum/cmd/utils/flags.go:98 +0x265
main.init()
	/private/tmp/ethermint-20180215-18815-12utxc0/ethermint-0.5.4/src/github.com/tendermint/ethermint/cmd/ethermint/main.go:18 +0x17a
```

## More info
`tendermint version`
0.15.0

`go version`
go version go1.9.4 darwin/amd64

`go env`
```
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/emizzle/go"
GORACE=""
GOROOT="/usr/local/Cellar/go/1.9.4/libexec"
GOTOOLDIR="/usr/local/Cellar/go/1.9.4/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/b3/v732nxrx2fn3vfmm_vwpm1v00000gn/T/go-build372967240=/tmp/go-build -gno-record-gcc-switches -fno-common"
CXX="clang++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
```

I am pretty new to all this, so apologies if I missed something. More than happy to provide more info. I am super keen to get this up and running and any help at all would be much appreciated!

Cheers,
Eric

**Comment by d10r** [comment]
Maybe you miss this:
`cp -a $GOPATH/src/github.com/tendermint/ethermint/setup/* ~/.ethermint/`

It's mentioned [here](http://ethermint.readthedocs.io/en/master/getting-started/getting-started.html#setup-ethermint), but not very explicitly (with command).

**Comment by emizzle** [comment]
Fixed, the issue was that by running `brew install` before installing from source, the symlinked binary from brew was always being run instead of my installed source executable. 

When installing via brew, a symlink was created in `/usr/local/bin/ethermint` pointing to to the brew cellar `/usr/local/Cellar/ethermint/0.5.4/bin/ethermint` and this was being run because `/usr/local/bin` has precedence in my $PATH before `$GOPATH/bin`. So running `ethermint` would always use the binary downloaded from brew, which, from what I can see, there may be an issue with. 

To see if you have the same issue, run `which ethermint` to see where your executable is running from. For me, I simply deleted the symlink `/usr/local/bin/ethermint` and that was enough. Probably a better option would be to put `$GOPATH/bin` before `/usr/local/bin` in $PATH.
